### PR TITLE
Actually emitting an error when any validation errors are found

### PR DESCRIPTION
**Pull request by ChrisMBarr**
This will let consumers know at a glance if anything was wrong.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ function validate(file, options, cb) {
           }
         }
       }
+      this.emit('error', new gutil.PluginError(pluginName, 'HTML Validation errors found!'));
       cb();
```
Comment (ewu02) [inline index.js]: How about pass error through callback? Something like:

```
cb('HTML Validation errors found!');
```

Should exit with non-zero error code

**Comment by ewu02** [comment]
That wouldn't work well with watchers as it would silently stop working. Can manually handle it or use plumber plugin with its errorHandler option. Passing error through callback should obviate the need for errorHandler option though. 

But even then, non-watcher tasks would terminate after the first validation error without processing remaining files in the stream. This wouldn't be ideal for tasks that like to treat validation errors as warnings.

Perhaps makes this optional?

**Comment by ChrisMBarr** [comment]
yes, making it optional wold be perfect, it's how i've seen many other gulp
validators handle this!

File: index.js (hunk 1 above)
Comment (ChrisMBarr) [inline index.js]: Yes, anything to be able to detect errors sound wonderful!

**Comment by ChrisMBarr** [comment]
I've tried numerous ways in gulp to figure out if any errors were found, and nothing can detect this currently.

I've tried stuff like

```
.pipe(htmlValidate(htmlValidateOpts).on("error", function(err){
   console.log("Error Detected", err);
})
```

but nothing seems to work.

**Comment by ChrisMBarr** [comment]
ok, I've just push the change you suggested.  Will this work?

**Comment by ewu02** [comment]
well the same concern still apply. Maybe an optional flag can be used to determine whether to give a warning or exit with non-zero code if validation fails.

**Comment by ChrisMBarr** [comment]
OK. how about this?  We allow for either via the options `emitError` and `errorInCallback` which both default to be turned off

File: index.js [hunk 2]
```diff
@@ -39,7 +40,14 @@ function validate(file, options, cb) {
...
       }
-      cb();
+      if(options.emitError){
+        this.emit('error', new gutil.PluginError(pluginName, errorMessage));
+      }
+      if(options.errorInCallback){
+        cb(errorMessage);
+      }else{
+        cb();
+      }
```
Comment (ewu02) [inline index.js:50]: can you add spacing to make it consistent with rest of the file? Other than that, this looks good.

**Comment by ChrisMBarr** [comment]
I'm not entirely sure if this is what you meant... if not please just merge it in and make the style changes you need to.

**Pull request merged**